### PR TITLE
Purge service

### PR DESCRIPTION
**Pull request by pedrokiefer**
Adds a Purge Service that watches pods on the k8s cluster and removes the load from the k8s control-plane when doing lots of purges per minute.

Some open points:
- [x] Should we filter pods by service-name? Probably yes. 
- [x] Improve error handling when some of the instances fails to purge.

### CODE REVIEW

File: internal/purge/api.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+	quit := make(chan os.Signal, 2)
+	signal.Notify(quit, os.Interrupt, syscall.SIGTERM)
+	select {
+	case <-quit:
+		p.Stop()
+	case <-p.shutdown:
+	}
+}
+
+func errorMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
```
Comment (wpjunior) [inline internal/purge/api.go:101]: @pedrokiefer could you extract the duplicated `errorMiddleware` to a common package ?

File: internal/purge/api.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+		if rpaas.IsNotFoundError(err) {
+			return &echo.HTTPError{Code: http.StatusNotFound, Message: err}
+		}
+		return err
+	}
+}
+
+func (p *purge) setupEcho() {
+	p.e.HideBanner = true
+	p.e.HTTPErrorHandler = func(err error, c echo.Context) {
```
Comment (wpjunior) [inline internal/purge/api.go:122]: @pedrokiefer could you extract the duplicated `HTTPErrorHandler` to a common package ?

File: internal/purge/cache.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+func (p *purge) PurgeCache(ctx context.Context, name string, args rpaas.PurgeCacheArgs) (int, error) {
+	if args.Path == "" {
+		return 0, rpaas.ValidationError{Msg: "path is required"}
+	}
+
+	pods, port, err := p.lister.ListPods(name)
+	if err != nil {
+		return 0, rpaas.NotFoundError{Msg: fmt.Sprintf("Failed to find pods: %v", err)}
+	}
+	// ToDo: better error handling (accumulate errors?)
```
Comment (wpjunior) [inline internal/purge/cache.go:78]: yeap @pedrokiefer, we could use a multi error accumulator, and errors.Wrapf to describe name of failed pod

File: internal/purge/api.go (hunk 1 above)
Comment (pedrokiefer) [reply internal/purge/api.go]: Done

File: internal/purge/api.go (hunk 2 above)
Comment (pedrokiefer) [reply internal/purge/api.go]: Done!

File: internal/purge/cache.go (hunk 3 above)
Comment (pedrokiefer) [reply internal/purge/cache.go]: Used `hashicorp/go-multierror` to accumulate errors and `errors.Wrapf` to add the pod ip address to the error message. Add the same behaviour to default purge API.

**Pull request merged**